src/ruby/lib/grpc/core/call.rb

```diff
@@ -0,0 +1,20 @@
+module GRPC
+  module Core
+    class Call
+      # call-seq:
+      # metadata = call.metadata
+      #
+      # Gets the metadata object saved the call.
+      attr_reader :metadata
+
+      # call-seq:
+      # call.metadata = metadata
+      #
+      # Saves the metadata hash on the call.
+      def metadata=(val)
+        raise TypeError, "expected Hash, got #{val.class}" if val && !val.is_a?(::Hash)
```

Comment:
@inkel changing to `unless val.is_a?(::Hash)` would not be the same logic.  The current conditional allows `nil` *or* `Hash` (and the tests specify that `nil` is an acceptable value).